kymatio/scattering1d/frontend/numpy_frontend.py

```diff
@@ -27,26 +27,26 @@ def scattering(self, x):
         S = scattering1d(x, self.backend, self.psi1_f, self.psi2_f,
                          self.phi_f, max_order=self.max_order, average=self.average, pad_left=self.pad_left,
                          pad_right=self.pad_right, ind_start=self.ind_start, ind_end=self.ind_end,
-                         oversampling=self.oversampling, out_type=self.out_type)
+                         oversampling=self.oversampling)
 
         if self.out_type == 'array':
+            S = self.backend.concatenate([x['coef'] for x in S])
             scattering_shape = S.shape[-2:]
             new_shape = batch_shape + scattering_shape
-
             S = S.reshape(new_shape)
         elif self.out_type == 'dict':
+            S = {x['n']: x['coef'] for x in S}
             for k, v in S.items():
                 # NOTE: Have to get the shape for each one since we may have
                 # average == False.
                 scattering_shape = v.shape[-2:]
                 new_shape = batch_shape + scattering_shape
-
                 S[k] = v.reshape(new_shape)
         elif self.out_type == 'list':
             for x in S:
+                x.pop('n')
```

Comment:
Why do we pop in the first place?

Reply:
hm ... you're right, we don't have to. I think that we did this to match the returned formats of core 2D. Note that the dictionaries in the `out_S` list do not have an `n` field.
https://github.com/kymatio/kymatio/blob/master/kymatio/scattering2d/core/scattering2d.py

I can't see another justification for what we're doing now in 1D, i.e. to redact metadata that could potentially be useful for the user. `n`can be seen as a "pointer" (or tuple of pointers) to the filterbank `psi1_f` (or tuple of filterbanks `(psi1_f, psi2_f)`)

i'd be in favor of not popping it away. that change is backwards compatible and very easy to maintain (we'd be _removing_ a for loop!) This change adds some new functionality; but it's one that the users already had access too in a different way, namely, `S.keys()`under `out_type="dict"`

TL;DR i can delete that line if that makes you happy!

Reply:
Lets do it in a separate repo. For now can you create an issue?